pkgs/by-name/mo/monophony/package.nix

```
@@ -0,0 +1,77 @@
+{ lib
+, fetchFromGitLab
+, python3Packages
+, wrapGAppsHook4
+, gst_all_1
+, gobject-introspection
+, gtk4
+, yt-dlp
+, libadwaita
+, libsoup_3
+, glib-networking
+,
+}:
+python3Packages.buildPythonApplication rec {
+  pname = "monophony";
+  version = "2.2.2";
+  format = "other";
+
+  sourceRoot = "source/source";
+  src = fetchFromGitLab {
+    owner = "zehkira";
+    repo = "monophony";
+    rev = "v${version}";
+    hash = "sha256-vAX9ac706fKVYW1jjgUJ2GvGRWeytYfgf7Lal+gEm04=";
+  };
+
+  propagatedBuildInputs = with python3Packages; [
+    brotli
+    gst-python
+    mutagen
+    ordered-set
+    pycryptodomex
+    pygobject3
+    requests
+    websockets
+    ytmusicapi
+    mpris-server
+  ];
+
+  nativeBuildInputs = [
+    gobject-introspection
+    wrapGAppsHook4
+    python3Packages.nuitka
+  ];
+
+  buildInputs =
+    [
+      libadwaita
+      libsoup_3
+      glib-networking
+    ]
+    ++ (with gst_all_1; [
+      gst-plugins-base
+      gst-plugins-good
+      gstreamer
+    ]);
+
+  makeFlags = [ "prefix=$(out)" ];
+
+  preFixup = ''
+     gappsWrapperArgs+=(
+    --prefix PYTHONPATH : "$PYTHONPATH"
+    --prefix LD_LIBRARY_PATH : "${libsoup_3.out}/lib"
+    --prefix PATH : "${lib.makeBinPath [yt-dlp]}"
+     )
```

```suggestion
     buildPythonPath "$pythonPath"
     gappsWrapperArgs+=(
    --prefix PYTHONPATH : "$program_PYTHONPATH"
    --prefix PATH : "$program_PATH"
    # needed for gstreamer https
    --prefix LD_LIBRARY_PATH : "${libsoup_3.out}/lib"
     )
```

i think using "$PYTHONPATH" is frowned upon as it adds unnecessary dependencies. buildPythonPath seems slightly better.

---

This is the only thing left, fixing rn